src/main/java/org/opensearch/geospatial/ip2geo/jobscheduler/Datasource.java

```diff
@@ -640,10 +656,10 @@ public UpdateStats(final StreamInput in) throws IOException {
 
         @Override
         public void writeTo(final StreamOutput out) throws IOException {
-            out.writeOptionalVLong(lastSucceededAt.toEpochMilli());
+            out.writeOptionalVLong(lastSucceededAt == null ? null : lastSucceededAt.toEpochMilli());
```

Comment:
same as for previous call to write optional, I think null value should be already handled

Reply:
Same for previous one. I call `.toEpochMillis()` method.